### PR TITLE
LED Support fixes

### PR DESCRIPTION
**Pull request by izzy84075**
Move the EEPROM setting out of the RGB enable flags, fix a possible logic error in the decay, get rid of a floating point multiply/divide, and rewrite the breathing effect ISR to support changing speed much better.

### CODE REVIEW

**Comment by chrislewisdev** [comment]
Thanks a lot for putting this together, izzy! I missed the initial notifications for this PR but I'll make sure to give it a closer look soon. I won't be able to test it on my keyboard for a week since I'm away from home, but in the meantime I'll update the led-support branch against master so that we can better see how this will integrate with the latest code.

I'm also in the middle of trying to get my master code merged into QMK (https://github.com/qmk/qmk_firmware/pull/3754) but if that gets merged before we finalise this PR we can always make another one with your updates :)

**Comment by chrislewisdev** [comment]
Alright, I've updated the `led-support` branch and fixed some obvious compilation issues.

The main things that will impact your PR will be the renaming of everything from `momentum` to `velocikey`, along with the use of the `VELOCIKEY_ENABLE` compile-time toggle, but all up it shouldn't be too much to update.

File: tmk_core/common/eeconfig.c [hunk 1]
```diff
@@ -22,8 +22,8 @@ void eeconfig_init(void)
 #endif
 #if defined(RGBLIGHT_ENABLE) || defined(RGB_MATRIX_ENABLE)
     eeprom_update_dword(EECONFIG_RGBLIGHT,      0);
-    eeprom_update_byte(EECONFIG_MOMENTUM,       0);
 #endif
+    eeprom_update_byte(EECONFIG_MOMENTUM,       0);
```
Comment (chrislewisdev) [inline tmk_core/common/eeconfig.c]: Nice catch on this one :)

File: quantum/momentum.c [hunk 2]
```diff
@@ -36,11 +36,11 @@ void momentum_decay_task() {
     //Decay a little faster at half of max speed
     if (typing_speed > TYPING_SPEED_MAX_VALUE / 2) typing_speed -= 1;
     //Decay even faster at 3/4 of max speed
-    if (typing_speed > TYPING_SPEED_MAX_VALUE / 4 * 3) typing_speed -= 3;
+    if (typing_speed > (TYPING_SPEED_MAX_VALUE / 4) * 3) typing_speed -= 1;
```
Comment (chrislewisdev) [inline quantum/momentum.c]: I'm keen to see how this impacts the feel of the speed transitions when I can test it out next week!

Comment (izzy84075) [reply quantum/momentum.c]: You might end up changing the -1 in the last stage back to -3, I just noticed the order of operations problem with the max/4*3 and figured the -1 would probably be more "consistent".

**Comment by izzy84075** [comment]
Working on rebasing this on the current led-support branch. Waiting for a build to finish to test with, and then I'll push up the changes.

**Comment by izzy84075** [comment]
Forgot to comment on it after doing so, but the changes to rebase on the current led-support branch were pushed up. Still running smoothly on my Tada68 at work.

**Comment by chrislewisdev** [comment]
Thanks izzy! I've had the chance to test this out on my DZ60 and I can see that there's no negative impacts of this change.

I'll merge this change in to the led-support branch, but I think I'm going to hold off on merging this to master for a little bit - my PR into QMK master has been taking a while to get across the line, so introducing this change might slow the review down even more. If it's OK with you, we could create a PR to merge this into QMK as soon as my main change is merged?

**Comment by izzy84075** [comment]
Yeah, that makes sense. Though it looks like they're understanding it as /only/ working with RGB right now, so they're thinking of it as being a subfeature of RGB. This PR would kinda flip that around, promoting it back out of the RGB stuff.

**Comment by chrislewisdev** [comment]
Hey @izzy84075, I know it took a while, but my QMK PR finally got merged 😆 

If you're still interested in getting the LED support into QMK, I'd suggest taking the work you did here and creating a new PR into QMK; I still don't have any LED boards to test this with, but I am of course happy to help with any code assistance to finalise it, if required :)

**Pull request merged**